modules/OCPP201/data_transfer/ocpp_data_transferImpl.cpp

```diff
@@ -16,10 +16,34 @@ void ocpp_data_transferImpl::ready() {
 
 types::ocpp::DataTransferResponse
 ocpp_data_transferImpl::handle_data_transfer(types::ocpp::DataTransferRequest& request) {
-    auto ocpp_response = mod->charge_point->data_transfer_req(request.vendor_id, request.message_id, request.data);
+    ocpp::v201::DataTransferRequest ocpp_request;
+    ocpp_request.vendorId = request.vendor_id;
+    if (request.message_id.has_value()) {
+        ocpp_request.messageId = request.message_id.value();
+    }
+    if (request.data.has_value()) {
+        ocpp_request.data = json::parse(request.data.value());
```

Comment:
id suggest to catch the exception for all `json::parse` in case the data could not be parsed